client/src/components/MultiJurisdictionAudit/AASetup/Review/index.tsx

```diff
@@ -139,6 +139,58 @@ const Review: React.FC<IProps> = ({
     isFileProcessed(j.ballotManifest)
   ).length
 
+  const getBatchTotal = () =>
+    contests
+      ? participatingJurisdictions.reduce(
+          (a, { ballotManifest: { numBatches } }) =>
+            numBatches !== null ? a + numBatches : a,
+          0
+        )
+      : 0
+
+  const getBallotTotal = (jurisdictionIds: string[]) =>
+    contests
+      ? jurisdictions
+          .filter(jurisdiction =>
+            jurisdictionIds.find(p => p === jurisdiction.name)
+          )
+          .reduce(
+            (a, { ballotManifest: { numBallots } }) =>
+              numBallots !== null ? a + numBallots : a,
+            0
+          )
+      : 0
+
+  const validateCustomSampleSize = (
+    totalBallotsCast: string,
+    jurisdictionIds: string[]
+  ) => {
+    if (auditType === 'BALLOT_POLLING') {
+      return testNumber(
+        Number(totalBallotsCast),
+        `Must be less than or equal to: ${totalBallotsCast} (the total number of ballots in this targeted contest)`
+      )
+    }
+    if (auditType === 'BATCH_COMPARISON') {
+      return testNumber(
+        Number(getBatchTotal()),
+        `Must be less than or equal to: ${Number(
+          getBatchTotal()
+        )} (the total number of batches in this targeted contest)`
+      )
+    }
+    if (auditType === 'BALLOT_COMPARISON') {
+      return testNumber(
+        Number(getBallotTotal(jurisdictionIds)),
```

Comment:
could you just use `totalBallotsCast` for this case as well?

Reply:
@jonahkagan It comes up `null` here when on Ballot Comparison audits.

Reply:
oh I see why that is. it gets set when computing the sample sizes, but since the api call for contest probably happens before the api call for sample sizes, it doesn't show up. side effects like that are tricky. i guess it's okay to keep what you have here, though it's not the same logic as the backend validation